### ISSUE TITLE
Send push to Users by tag

### ISSUE THREAD

**Issue by aconital** [laravel-notification-channels/onesignal#47]
In OneSignal I have added tags for my users like this `{email: Myemail@gmail.com}`. How can I send push to users based on that tag? I tried this but it doesn't work.
`
public function routeNotificationForOneSignal()
{
    return ['email' => 'Myemail@gmail.com'];
}`

**Comment by LKDevelopment** [comment]
Actually you can't send by tags, only per User. With
`public function routeNotificationForOneSignal() { return ['email' => 'Myemail@gmail.com']; }`
you use the syned Hash Mail Function from OneSignal

**Comment by aconital** [comment]
@LKDevelopment it would be really nice if this feature is added. This library handles it https://github.com/berkayk/laravel-onesignal

**Comment by LKDevelopment** [comment]
@aconital I've added it in the Branch: https://github.com/laravel-notification-channels/onesignal/tree/implement-targetting-per-tags Could you please check it out and test it?

**Comment by aconital** [comment]
@LKDevelopment awesome thanks.

**Comment by LKDevelopment** [comment]
Could you please test it? When you say it works we will publish the 1.2.0 which will contains this :)

**Comment by aconital** [comment]
@LKDevelopment it worked just tested.

**Comment by aconital** [comment]
@LKDevelopment Although I think this is not a scalable solution having the tag filter hardcoded in the `routeNotificationForOneSignal` function. What if I want to send 2 different types of notification with different rules? My suggestion is to get rid of routeNotificationForOneSignal function and incorporate it into the custom Notifcation class as a function

**Comment by Lloople** [comment]
Hey @aconital, can you send a PR with that code applied?

**Comment by kjostling** [comment]
Hi, Can´t get tags to work like the docs say it should (not the email as discussed above). I´ve tried some variations of this:
```
    public function routeNotificationForOneSignal()
    {
        return ['tags' => ['key' => 'userId', 'relation' => '=', 'value' => $this->uuid]];
    }
```
All I get back is "include_player_ids must be an array" from OneSignal. I´ll keep testing, see if I figure it out.

**Comment by LKDevelopment** [comment]
@kjostling the version that contains sending via Tags isn't released yet, because i've tested it a bit on a real world application. Since this looks good i will release the update today.

**Comment by LKDevelopment** [comment]
Released as 1.2.0.
